pkg/deploy/taskdir/definitions/def_0_3.go

```
@@ -168,6 +198,16 @@ func (d *GoDefinition_0_3) fillInUpdateTaskRequest(ctx context.Context, client a
 	return nil
 }
 
+func (d *GoDefinition_0_3) hydrateFromTask(ctx context.Context, client api.IAPIClient, t *api.Task) error {
+	d.Arguments = t.Arguments
+	if v, ok := t.KindOptions["entrypoint"]; ok {
+		if sv, ok := v.(string); ok {
```

what if it's not a string? is that an error case? or expected?

---

Same in the other places

---

hmm it's probably an error case

---

Let's throw an error then because if we don't, we will silently fail - the user will think their task def is valid and working but we're actually silently skipping fields!

---

Although we should already hit this error when we convert the yaml -> go.... but I think we should add errors here anyways